lib/statsig_configs.ex

```diff
@@ -0,0 +1,136 @@
+defmodule Statsig.Configs do
+  use GenServer
+  require Logger
+
+  @table_name :statsig_configs
+
+  @type state :: %{
+    reload_interval: integer(),
+    last_sync_time: integer(),
+    reload_timer: reference() | nil
+  }
+
+  def start_link(_) do
+    GenServer.start_link(__MODULE__, %{
+      reload_interval: 10_000,
+      last_sync_time: 0,
+      reload_timer: nil,
+    }, name: __MODULE__)
+  end
+
+  def initialize(reload_interval) do
+    GenServer.call(__MODULE__, {:initialize, reload_interval})
+  end
+
+  def shutdown do
+    GenServer.call(__MODULE__, :shutdown)
+  end
+
+  @impl true
+  def init(state) do
+    :ets.new(@table_name, [:named_table, :set, :public])
+    {:ok, state}
+  rescue
+    ArgumentError ->
+      {:ok, state}
+  end
+
+  @impl true
+  def handle_call({:initialize, reload_interval}, _from, state) do
+    new_state = if is_number(reload_interval) do
```

Comment:
you can get rid of this check by using a guard
```
```suggestion
  def handle_call({:initialize, reload_interval}, _from, state) 
     when is_integer(reload_interval) do
```